### PR TITLE
Added #get_node, #node_exists? methods on Graph and relevant tests. 

### PR DESCRIPTION
**Pull request by RedFred7**
Added method to return Node based on node name. Added method to check if Node exists in the graph. Also changed indentation to match Ruby standard. Fixed minor bug in demo.rb.

### CODE REVIEW

File: spec/graphviz/graph_spec.rb [hunk 1]
```diff
@@ -20,13 +20,13 @@
 
 require 'graphviz'
 
-  SAMPLE_GRAPH_DOT = <<~EOF
-  digraph "G" {
+  SAMPLE_GRAPH_DOT = <<-EOF
```
Comment (ioquatix) [inline spec/graphviz/graph_spec.rb]: Is this an improvement?

**Comment by ioquatix** [comment]
Please don't reformat the entire code base. Especially don't do it with functional code changes :)

File: spec/graphviz/graph_spec.rb (hunk 1 above)
Comment (RedFred7) [reply spec/graphviz/graph_spec.rb]: This was necessary for the gem to run on versions earlier than 2.3. The squiggly `<<~` HEREDOC syntax was introduced in Ruby 2.3 and previous versions will raise error when <<~ is used.

Comment (ioquatix) [reply spec/graphviz/graph_spec.rb]: Fair enough. I'm not actually too concerned about that since Ruby < 2.3 is EOL.

**Comment by ioquatix** [comment]
Okay, this looks good. I will review in more detail after merging.

**Comment by ioquatix** [comment]
Okay I released it as v1.1.0

**Pull request merged**